synapse/rest/admin/users.py

```diff
@@ -94,10 +94,10 @@ def __init__(self, hs):
         guests = parse_boolean(request, "guests", default=True)
         deactivated = parse_boolean(request, "deactivated", default=False)
 
-        users = await self.store.get_users_paginate(
+        users, total = await self.store.get_users_paginate(
             start, limit, user_id, guests, deactivated
         )
-        ret = {"users": users}
+        ret = {"data": users, "total": total}
```

Comment:
As far as I know, we're the only users of the new API. At first I wanted to mark it "draft", as the interface is for sure not stable yet, but I was told that kind of folder structure is outdated. Maybe we should mark it draft in the code and readme.
The main reason for this change is to have a generic API for users and rooms. We're currently using `react-admin` to create the admin UI and this would keep workarounds in the data adapter to a minimum.

Reply:
We can't assume that I'm afraid. If the API exists then people may have already started using it. The correct behaviour here now is to create a new API on `/_synapse/admin/v2/...` with the new semantics, and deprecate the old one.

Reply:
This is only for the draft v2 API.

Reply:
Ah, sorry, this path is already `v2`, yes. But the problem here is there's no indication in [the documentation](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst) that this is an unstable API.

You say you wanted to mark it as a draft. The best place for that probably would've been in the admin api docs. I apologise that you weren't given clear instructions there.

However since this has gone out in a stable release of Synapse already, it's too late to mark it as unstable retroactively. If we're going to change it in a non-backward-compatible way, we're going to need to bump the version number (in this case, to v3).

I'm also not entirely convinced we need to make this change just to make client code a little simpler.